tests.py

```diff
@@ -66,69 +66,73 @@ def testParseArguments(self):
 
   @classmethod
-  def setUpClass(cls):
+  def tearDownClass(cls):
+    shutil.rmtree(os.path.join('test_data', 'docker'))
+
+  @classmethod
+  def _setup(cls, driver, driver_class):
+    """Internal method to set up the TestCase on a specific storate."""
+    cls.driver = driver
     docker_directory_path = os.path.join('test_data', 'docker')
     if not os.path.isdir(docker_directory_path):
-      docker_tar = os.path.join('test_data', 'aufs.tgz')
+      docker_tar = os.path.join('test_data', cls.driver+'.tgz')
       tar = tarfile.open(docker_tar, 'r:gz')
       tar.extractall('test_data')
       tar.close()
+    cls.de_object = de.DockerExplorer()
+    cls.de_object._SetDockerDirectory(docker_directory_path)
 
-    de_test_object = de.DockerExplorer()
-    de_test_object.docker_directory = docker_directory_path
-    de_test_object.DetectStorage()
-    cls.storage = de_test_object.storage_object
-    cls.container_id = (
-        '7b02fb3e8a665a63e32b909af5babb7d6ba0b64e10003b2d9534c7d5f2af8966')
-    cls.container = cls.storage.GetContainer(cls.container_id)
-    cls.image_id = (
-        '7968321274dc6b6171697c33df7815310468e694ac5be0ec03ff053bb135e768')
-
-  @classmethod
-  def tearDownClass(cls):
-    shutil.rmtree(os.path.join('test_data', 'docker'))
+    cls.driver_class = driver_class
 
   def testDetectStorage(self):
-    """Tests the DockerExplorer.DetectStorage function in a AUFS storage."""
-    de_test_object = de.DockerExplorer()
-    de_test_object.docker_directory = 'this_dir_shouldnt_exist'
-
-    expected_error_message = (
-        'this_dir_shouldnt_exist is not a Docker directory\n'
-        'Please specify the Docker\'s directory path.\n'
-        'hint: de.py -r /var/lib/docker')
-
-    with self.assertRaises(errors.BadStorageException) as err:
-      de_test_object.DetectStorage()
-    self.assertEqual(expected_error_message, err.exception.message)
-
-    de_test_object.docker_directory = os.path.join('test_data', 'docker')
-    de_test_object.DetectStorage()
-    storage_object = de_test_object.storage_object
+    """Tests the DockerExplorer.DetectStorage function."""
+    storage_object = self.de_object.storage_object
     self.assertIsNotNone(storage_object)
-    self.assertIsInstance(storage_object, aufs.AufsStorage)
-    self.assertEqual(storage_object.STORAGE_METHOD, 'aufs')
+    self.assertIsInstance(storage_object, self.driver_class)
+    self.assertEqual(storage_object.STORAGE_METHOD, self.driver)
 
     self.assertEqual(2, storage_object.docker_version)
     self.assertEqual('config.v2.json',
-                     storage_object.container_config_filename)
+                     self.de_object.container_config_filename)
+
+
+class TestAufsStorage(StorageTestCase):
+  """Tests methods in the Storage object."""
+
+  @classmethod
+  def setUpClass(cls):
+    cls._setup('aufs', aufs.AufsStorage)
 
   def testGetAllContainers(self):
     """Tests the GetAllContainers function on a AuFS storage."""
```

Comment:
I'm not super familiar with testing but it seems this test case bundles a bunch of different tests: testing name, timestamp, image name, and length of the returned items

Reply:
Yes.
I "test" one function per test here. Probably not the best.
filed #24 to deal with this later